Formula/pinentry-mac.rb

```diff
@@ -28,6 +29,20 @@ class PinentryMac < Formula
     sha256 "d4bcf2003fa1345ecb1809461140179a3737e8e03eb49d623435beb3c2f09b64"
   end
 
+  patch do
+    # Fix implicit function declaration (build error for Xcode 12+)
+    # https://github.com/GPGTools/pinentry-mac/pull/8
+    url "https://github.com/GPGTools/pinentry-mac/commit/4c99326362df729f05ebee32c331a30de9437e0c.patch?full_index=1"
+    sha256 "6ef489d62ac729fdb3098354ab86ed32603f602d32dfb163c813b17c9d64f040"
+  end
+
+  patch do
+    # Add arm64 to build targets
+    # https://github.com/GPGTools/pinentry-mac/pull/9
+    url "https://github.com/GPGTools/pinentry-mac/commit/cc38341adc179a26ac3ea3754a4013afb9681930.patch?full_index=1"
+    sha256 "814294a2b0a183dbde31eecd0ab58208c0d91c3e458a45a13aa04b0ebd17389b"
+  end
```

Comment:
```suggestion
  if Hardware::CPU.arm?
    patch do
      # Fix implicit function declaration (build error for Xcode 12+)
      # https://github.com/GPGTools/pinentry-mac/pull/8
      url "https://github.com/GPGTools/pinentry-mac/commit/4c99326362df729f05ebee32c331a30de9437e0c.patch?full_index=1"
      sha256 "6ef489d62ac729fdb3098354ab86ed32603f602d32dfb163c813b17c9d64f040"
    end

    patch do
      # Add arm64 to build targets
      # https://github.com/GPGTools/pinentry-mac/pull/9
      url "https://github.com/GPGTools/pinentry-mac/commit/cc38341adc179a26ac3ea3754a4013afb9681930.patch?full_index=1"
      sha256 "814294a2b0a183dbde31eecd0ab58208c0d91c3e458a45a13aa04b0ebd17389b"
    end
  end
```

Your patch broke all the Intel builds so it may need to be applied selectively.